server/backends/migration_cache/config.go

```diff
@@ -0,0 +1,37 @@
+package migration_cache
+
+import (
+	"time"
+
+	"github.com/buildbuddy-io/buildbuddy/server/util/disk"
+)
+
+type MigrationConfig struct {
+	Src  *CacheConfig `yaml:"src"`
+	Dest *CacheConfig `yaml:"dest"`
+}
+
+type CacheConfig struct {
+	DiskConfig   *DiskCacheConfig   `yaml:"disk"`
+	PebbleConfig *PebbleCacheConfig `yaml:"pebble"`
+}
```

Comment:
Idea: make this an interface that defines "CacheFromConfig() interfaces.Cache`, and then you don't have to validate the CacheConfig object.

Reply:
Can you say more about how to do that in a way where we can still enforce the structure of what we're expecting of a DiskCacheConfig, for example?

Reply:
Going to merge this for now bc that will make some other development easier, but can come back and refactor this

Reply:
The idea is that you make `DiskCacheConfig` implement `CacheConfig`, and then whatever gets passed in as a `CacheConfig` is guaranteed to just be one concrete type of `CacheConfig`, so it doesn't need to be validated. Also if we want to add a new type later, it simply has to implement `CacheConfig` and there are no changes that need to be made elsewhere, either in validation or in `CacheConfig`'s (now non-existent) fields.

Reply:
I like the concept but does that work with the JSON struct reflection you helped me with last week? We'd need the yaml parser to figure out which concrete type the interface needs to be cast to right?

Reply:
I tried this, which is what I think you're suggesting: https://github.com/buildbuddy-io/buildbuddy/pull/2507/files

Getting errors like: 	panic: reflect.Set: value of type map[string]interface {} is not assignable to type migration_cache.CacheConfig